ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/computation/ValidatedCommandFactory.java

```diff
@@ -61,19 +57,56 @@ public Command create(
    * @param context The KSQL engine snapshot to validate the command against.
    * @return A validated command, which is safe to enqueue onto the command topic.
    */
+  @SuppressWarnings("MethodMayBeStatic") // Not static to allow dependency injection
   public Command create(
       final ConfiguredStatement<? extends Statement> statement,
       final ServiceContext serviceContext,
-      final KsqlExecutionContext context) {
+      final KsqlExecutionContext context
+  ) {
+    return ensureDeserializable(createCommand(statement, serviceContext, context));
+  }
+
+  /**
+   * Ensure any command written to the command topic can be deserialized.
+   *
+   * <p>Any command that can't be deserialized is a bug. However, given a non-deserializable
+   * command will kill the command runner thread, this is a safety net to ensure commands written to
+   * the command topic can be deserialzied.
```

Comment:
I suppose this isn't _strictly_ true if the version reading is different than the version writing; but I'm not opposed to the extra safety net 😛